crates/qasm3/src/build.rs

```diff
@@ -0,0 +1,406 @@
+                if is_const.clone().into() {
+                    Err(QASM3ImporterError::new_err("cannot handle consts"))
+                } else if decl.initializer().is_some() {
+                    Err(QASM3ImporterError::new_err(
+                        "cannot handle initialised bits",
+                    ))
+                } else {
+                    self.add_clbit(py, name_id.clone())
+                }
+            }
+            Type::BitArray(dims, is_const) => {
+                if is_const.clone().into() {
+                    Err(QASM3ImporterError::new_err("cannot handle consts"))
+                } else if decl.initializer().is_some() {
+                    Err(QASM3ImporterError::new_err(
+                        "cannot handle initialised registers",
+                    ))
+                } else {
+                    match dims {
+                        ArrayDims::D1(size) => {
+                            self.add_creg(py, name_id.clone(), name_symbol.name(), *size)
+                        }
+                        _ => Err(QASM3ImporterError::new_err(
+                            "cannot handle quantum registers with more than one dimension",
+                        )),
+                    }
+                }
+            }
+            ty => Err(QASM3ImporterError::new_err(format!(
+                "unhandled classical type: {:?}",
+                ty,
+            ))),
+        }
+    }
+
+    fn declare_quantum(
+        &mut self,
+        py: Python,
+        ast_symbols: &SymbolTable,
+        decl: &asg::DeclareQuantum,
+    ) -> PyResult<()> {
+        let name_id = decl
+            .name()
+            .as_ref()
+            .map_err(|err| QASM3ImporterError::new_err(format!("internal error {:?}", err)))?;
+        let name_symbol = &ast_symbols[name_id];
+        match name_symbol.symbol_type() {
+            Type::Qubit => self.add_qubit(py, name_id.clone()),
+            Type::QubitArray(dims) => match dims {
+                ArrayDims::D1(size) => {
+                    self.add_qreg(py, name_id.clone(), name_symbol.name(), *size)
+                }
+                _ => Err(QASM3ImporterError::new_err(
+                    "cannot handle quantum registers with more than one dimension",
+                )),
+            },
+            _ => unreachable!(),
+        }
+    }
+
+    fn call_gate(
+        &mut self,
+        py: Python,
+        ast_symbols: &SymbolTable,
+        call: &asg::GateCall,
+    ) -> PyResult<()> {
+        if call.modifier().is_some() {
+            return Err(QASM3ImporterError::new_err("gate modifiers not handled"));
+        }
+        let gate_id = call
+            .name()
+            .as_ref()
+            .map_err(|err| QASM3ImporterError::new_err(format!("internal error {:?}", err)))?;
+        let gate = self.symbols.gates.get(gate_id).ok_or_else(|| {
+            QASM3ImporterError::new_err(format!("internal logic error: unknown gate {:?}", gate_id))
```

Comment:
No, I can make them all consistent.

Reply:
Oh, actually looking now, there is a bit of a difference - I used "internal error" in situations where I have to handle a case/variant from `oq3_semantics` that _should_ be unreachable, and "internal logic error" if I had to handle a branch that (in theory, at least) would have been an error on this side.  Not quite sure why some places panic rather than raising the correct error, though.

Anyway, I just made them all "internal error" and turned them into proper exceptions in a2d2c4f3ec